app/Http/Controllers/BillingController.php

```diff
@@ -662,4 +667,29 @@ private function formatHistoricalUsageData($historicalUsageData)
         return $formattedData;
     }
 
+    private function paginate($items, $perPage, $setDefaultOption = true, $options = [])
+    {
+        if($setDefaultOption){
+            $options = ['path' => request()->url(), 'query' => request()->query()];
+        }
+
+        $requestUrl = $this->cleanUrl(request()->url());
+
+        if (isset($options['path']) && env('NGINX_HOST') . $options['path'] == $requestUrl) {
+            $page = Input::get('page', 1); // Get the current page or default to 1
+        } else {
+            $page = 1; // Get the current page or default to 1
+        }
+
+
+        $items = $items instanceof Collection ? $items : Collection::make($items);
+
+        return new LengthAwarePaginator($items->forPage($page, $perPage), $items->count(), $perPage, $page, $options);
+    }
+
+    private function cleanUrl($url)
```

Comment:
Why are you replacing https:// URLs with http:// ?

Reply:
This is actually a one line to replace `https://` and `http://`. Tested and this logic works in regards to remove both `https://` and `http://`